### PR TITLE
bitnami/kafka Add extraListeners to kafka.listeners in _helpers.tpl

### LINKED ISSUE #21746: listeners are not configured or appended properly when am trying to enable extraListeners
### Name and Version

bitnami/kafka --version 23.0.2 App version 3.5.0

### What architecture are you using?

None

### What steps will reproduce the bug?

I am deploying three node kafka-kraft version in kubernets environment is working fine.
in my use case need one more listener to access the broker externally, based on the documentation added the extra listeners configuration.
tried 
case-1
listeners:
  extraListeners:
    extntls:
      containerPort: 9097
      protocol: PLAINTEXT
      name: EXTNTLS
      sslClientAuth: ""  
case-2
listeners:
  extraListeners:
    - name: EXTNTLS
      containerPort: 9097
      protocol: SASL_PLAINTEXT
      sslClientAuth: ""  

case-3
listeners:
  client:
    containerPort: 9092
    protocol: SASL_PLAINTEXT
    name: CLIENT
    sslClientAuth: ""
    
  controller:
    name: CONTROLLER
    containerPort: 9093
    protocol: SASL_PLAINTEXT
    sslClientAuth: ""

  interbroker:
    containerPort: 9094
    protocol: SASL_PLAINTEXT
    name: INTERNAL
    sslClientAuth: ""

  extraListeners:
    - name: EXTNTLS
      containerPort: 9097
      protocol: SASL_PLAINTEXT
      sslClientAuth: ""    

 

### Are you using any custom parameters or values?

Custom values mentioned in case

### What is the expected behavior?

KAFKA_CFG_LISTENERS : INTERNAL://:9094,CLIENT://:9092,CONTROLLER://:9093,EXTERNAL://:9095, EXTNTLS://:9097 with append with extra listener

Client able access the broker externally with those two ports

### What do you see instead?

KAFKA_CFG_LISTENERS : map[extraListeners:map[externalntls:map[containerPort:9098 name:EXTERNALNTLS protocol:PLAINTEXT sslClientAuth:]]]

### Additional information

kindly review and suggest, is there any issue with particular version (--version 23.0.2 App version 3.5.0) chart or we are doing anything wrong while configuring extraListeners ?

### LINKED ISSUE #21747: listeners are not configured or appended properly when am trying to enable extraListeners
### Name and Version

bitnami/kafka 24.0.4

### What architecture are you using?

None

### What steps will reproduce the bug?

I am deploying three node kafka-kraft version in K8s environment is working fine.
presently require one more listener to access the broker externally, based on the documentation added the extra listeners configuration.
tried 
case-1
```yaml
listeners:
  extraListeners:
    extntls:
      containerPort: 9097
      protocol: PLAINTEXT
      name: EXTLS
      sslClientAuth: ""  
```
case-2
```yaml
listeners:
  extraListeners:
    - name: EXTNTLS
      containerPort: 9097
      protocol: PLAINTEXT
      sslClientAuth: ""  
```
case-3
```yaml
listeners:
  client:
    containerPort: 9092
    protocol: PLAINTEXT
    name: CLIENT
    sslClientAuth: ""
    
  controller:
    name: CONTROLLER
    containerPort: 9093
    protocol: SASL_PLAINTEXT
    sslClientAuth: ""

  interbroker:
    containerPort: 9094
    protocol: PLAINTEXT
    name: INTERNAL
    sslClientAuth: ""

  extraListeners:
    - name: EXTLS
      containerPort: 9097
      protocol: PLAINTEXT
      sslClientAuth: ""    
```

### Are you using any custom parameters or values?

Custom values mentioned in case

### What is the expected behavior?

`KAFKA_CFG_LISTENERS : INTERNAL://:9094,CLIENT://:9092,CONTROLLER://:9093,EXTERNAL://:9095, EXTNTLS://:9097` with append with extra listener

Client able access the broker externally with those two ports

### What do you see instead?

Formatting storage directories to add metadata...
```
Exception in thread "main" java.lang.IllegalArgumentException: requirement failed: advertised.listeners listener names must be equal to or a subset of the ones defined in listeners. Found CLIENT,INTERNAL,EXTLS,EXTERNAL. The valid options based on the current configuration are CLIENT,INTERNAL,EXTERNAL,CONTROLLER
	at scala.Predef$.require(Predef.scala:281)
	at kafka.server.KafkaConfig.validateValues(KafkaConfig.scala:2317)
	at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:2160)
	at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:1568)
	at kafka.tools.StorageTool$.$anonfun$main$1(StorageTool.scala:50)
	at scala.Option.flatMap(Option.scala:271)
	at kafka.tools.StorageTool$.main(StorageTool.scala:50)
	at kafka.tools.StorageTool.main(StorageTool.scala)
```
### Additional information

kindly review and suggest, is there any issue with particular version(--version 24.0.4 App version 3.5.1) chart or are we doing anything wrong while configuring extra Listeners and is that possible to expose two ports externally each having different security protocol (PLAINTEXT, SSL)?

### PR DESCRIPTION
**Pull request by Voldemat**
### Description of the change
Changed kafka chart _helpers.tpl
Added extraListeners to kafka.listeners the same way it is done in kafka.advertisedListeners

### Benefits
Allow to use extraListeners without manually overriding them.

### Possible drawbacks

None, as without it kafka doesn`t start.
Exception in thread "main" java.lang.IllegalArgumentException: requirement failed: advertised.listeners listener names must be equal to or a subset of the ones defined in listeners. Found CLIENT,INTERNAL,CLUSTER,EXTERNAL. The valid options based on the current configuration are CLIENT,INTERNAL,EXTERNAL,CONTROLLER

### Applicable issues
- fixes #21747
- fixes #21746

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by Voldemat** [comment]
Also fixed and error with openssl when using pem auth with custom key. Just provided necessary password from env

File: bitnami/kafka/templates/scripts-configmap.yaml [hunk 1]
```diff
@@ -192,15 +192,15 @@ data:
         if [[ -f "/mounted-certs/kafka-${POD_ROLE}-${POD_ID}.crt" && "/mounted-certs/kafka-${POD_ROLE}-${POD_ID}.key" ]]; then
           cp "/mounted-certs/kafka-${POD_ROLE}-${POD_ID}.crt" /certs/tls.crt
           # Copy the PEM key ensuring the key used PEM format with PKCS#8
-          openssl pkcs8 -topk8 -nocrypt -in "/mounted-certs/kafka-${POD_ROLE}-${POD_ID}.key" > /certs/tls.key
+          openssl pkcs8 -topk8 -nocrypt -passin pass:"${KAFKA_TLS_PEM_KEY_PASSWORD}" -in "/mounted-certs/kafka-${POD_ROLE}-${POD_ID}.key" > /certs/tls.key
```
Comment (juan131) [inline bitnami/kafka/templates/scripts-configmap.yaml:195]: This change isn't related with the extra listeners topic. Please create a separate PR for this.

File: bitnami/kafka/templates/_helpers.tpl [hunk 2]
```diff
@@ -459,6 +459,9 @@ Returns the Kafka listeners settings based on the listeners.* object
   {{- $listeners = append $listeners .context.Values.listeners.controller -}}
   {{- end -}}
   {{- end -}}
+  {{- range $i := .Values.listeners.extraListeners -}}
```
Comment (juan131) [inline bitnami/kafka/templates/_helpers.tpl:462]: Missing `.context`:
```suggestion
  {{- range $i := .context.Values.listeners.extraListeners -}}
```

Comment (Voldemat) [reply bitnami/kafka/templates/_helpers.tpl]: fixed

File: bitnami/kafka/templates/scripts-configmap.yaml (hunk 1 above)
Comment (Voldemat) [reply bitnami/kafka/templates/scripts-configmap.yaml]: Here is a new PR #28618

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**